cpp/CMakeLists.txt

```diff
@@ -21,14 +21,16 @@ set(COMMON_INCLUDES ${PROJECT_SOURCE_DIR})
 include_directories(${EIGEN_INCLUDE_DIRS})
 include_directories(${COMMON_INCLUDES}) 
 
-# Set source files
-file(GLOB_RECURSE SOURCES RELATIVE ${CMAKE_SOURCE_DIR} src/*.cc)
-message(STATUS "Source files: " ${SOURCES})
 
 # Locate CUDA package 
 find_package(CUDA)
 if (CUDA_FOUND)
 
+  # Set source files
+  file(GLOB_RECURSE SOURCES RELATIVE ${CMAKE_SOURCE_DIR} src/*.cc src/core/gpu_util/*.cu)
```

Comment:
I see why you change the CMakeLists.txt this way. But I think the gpu util functions are not kernels so I would suggest use .cc for gpu util functions instead and change CMakeLists.txt to what it was like previously. To let it compile successfully in CPU-only machine, use macro definition "NEED_CUDA" like that in the gpu_util.h